app/com/arpnetworking/metrics/portal/scheduling/TwoWayJobRefSerializer.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.metrics.portal.scheduling;
+
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.google.common.collect.ImmutableMap;
+import models.internal.impl.DefaultOrganization;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+
+/**
+ * A {@link JobRefSerializer} that allows construction to and from entity IDs.
+ *
+ * Round-trip is only guaranteed for refs whose repository types are whitelisted by this serializer
+ * (e.g. allowed by the constructor arguments).
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public class TwoWayJobRefSerializer implements JobRefSerializer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TwoWayJobRefSerializer.class);
+
+    private final Map<String, Class<? extends JobRepository<?>>> _repositories;
+    private final Map<String, Class<? extends JobExecutionRepository<?>>> _execRepositories;
```

Comment:
I'm not entirely sure this whitelisting is necessary, but I thought to be safe we should avoid constructing a reference to any class we haven't vetted (configured) ourselves.

Reply:
See comment above. I think there are some hidden complexities here that do not have high ROI now.

Reply:
I am more concerned about ActorA moving from HostA to HostB which have independent locally cloned repositories and say one is not yet open. Such a race condition would result in actor death.

Reply:
> I think there are some hidden complexities here that do not have high ROI now

If that's the case, I have any qualms with removing this – simpler is better.

> I am more concerned about ActorA moving from HostA to HostB which have independent locally cloned repositories and say one is not yet open

Sure, but I think that concern is orthogonal to restarts in the general sense.